autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/TabularColumnTransformer.py

```diff
@@ -48,13 +48,16 @@ def fit(self, X: Dict[str, Any], y: Any = None) -> "TabularColumnTransformer":
             "TabularColumnTransformer": an instance of self
         """
         self.check_requirements(X, y)
-        numerical_pipeline = 'drop'
-        categorical_pipeline = 'drop'
+        # in case the preprocessing steps are disabled
+        # i.e, NoEncoder for categorical, we want to
+        # let the data in categorical columns pass through
+        numerical_pipeline = 'passthrough'
+        categorical_pipeline = 'passthrough'
 
         preprocessors = get_tabular_preprocessers(X)
-        if len(X['dataset_properties']['numerical_columns']):
+        if len(X['dataset_properties']['numerical_columns']) and len(preprocessors['numerical']):
             numerical_pipeline = make_pipeline(*preprocessors['numerical'])
-        if len(X['dataset_properties']['categorical_columns']):
+        if len(X['dataset_properties']['categorical_columns']) and len(preprocessors['categorical']):
```

Comment:
These lines make no sense to me, `if len(...) and <cond>`, I"m assuming it's checking for `len(...) > 0` but I would prefer to be explicit about that.

Reply:
yeah, I agree. actually, we don't need to check for the length of columns as we will only have preprocessors if the length of columns is > 0. So, I'll change these conditions.